src/value.rs

```diff
@@ -72,6 +72,10 @@ pub enum Value {
     #[cfg(feature = "with-bigdecimal")]
     #[cfg_attr(docsrs, doc(cfg(feature = "with-bigdecimal")))]
     BigDecimal(Option<Box<BigDecimal>>),
+
+    #[cfg(feature = "with-array")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "with-array")))]
+    Array(Option<Box<Vec<Value>>>),
```

Comment:
Do you think a solution like this will work out:

```
pub enum Value<A> {
```

and then this becomes:

```
    Array(Option<Box<Vec<A>>>),
```

We would have to convert the other code to account for the generic argument, but I'm thinking if such a approach will work.

Reply:
> ```
> pub enum Value<A> {
> ```

So, `A` is unbounded? Why is this necessary?